core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1061,6 +1072,7 @@ class ReplicaManager(val config: KafkaConfig,
       logReadResultMap.put(topicIdPartition, logReadResult)
     }
 
+    val delayedRemoteFetchPurgatory: DelayedOperationPurgatory[DelayedRemoteFetch] = null;
```

Comment:
Why should we make `delayedRemoteFetchPurgatory` null here? We've init `delayedRemoteFetchPurgatory` in L217, and want to use it below L1131 immediately. So this line should be removed.

Reply:
OK, I've run a test, and confirmed this line will make it throw NPE while fetching data from remote segment.